### PR TITLE
Boot notif response impl

### PR DESCRIPTION
**Pull request by tvelu77**
Boot notification response works with OCPP 1.6 and 2.0

### CODE REVIEW

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java [hunk 1]
```diff
@@ -40,22 +40,17 @@ public ChargePointManager(OcppVersion ocppVersion,
    *
    * @param webSocketRequestMessage WebSocketRequestMessage.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java:41]: missing a return doc

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java [hunk 2]
```diff
@@ -40,22 +40,17 @@ public ChargePointManager(OcppVersion ocppVersion,
...
-              60,
-              RegistrationStatus.Accepted);
+      var resp = ocppMessageBuilder.buildMessage(webSocketRequestMessage);
       messageSender.sendMessage(new WebSocketResponseMessage(3,
               webSocketRequestMessage.messageId(),
               JsonParser.objectToJsonString(resp)));
+      return JsonParser.objectToJsonString(resp);
     } catch (IllegalArgumentException e) {
       System.out.println(e.getMessage());
+      return "ERROR";
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java:53]: what about that, what do we get from this return ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java [hunk 3]
```diff
@@ -58,7 +58,10 @@ public void onMessage(WebSocket conn, String message) {
             + ": "
             + message);
     var webSocketMessage = WebSocketRequestMessage.parse(message);
-    chargePoints.get(conn.getRemoteSocketAddress()).processMessage(webSocketMessage);
+    LOGGER.info("sent message to "
+            + conn.getRemoteSocketAddress()
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java:62]: conn.getRemoteSocketAddress() in a variable ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp16.OcppMessageBuilder16;
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.OcppMessageBuilder2;
+import java.util.Objects;
+
+/**
+ * Interface used to define an OCPP message builder.
+ */
+public interface OcppMessageBuilder {
+
+  OcppMessage buildMessage(WebSocketRequestMessage webSocketRequestMessage);
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java:13]: javadoc ? those method are public by default

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+  String transform(OcppMessage message);
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java:15]: java doc

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * Builds an OCPP 1.6 message.
+ */
+public class OcppMessageBuilder16 implements OcppMessageBuilder {
+
+  @Override
+  public OcppMessage buildMessage(WebSocketRequestMessage webSocketRequestMessage) {
+    return switch (webSocketRequestMessage.messageName()) {
+      case "BootNotification" ->
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java:18]: "BootNotification" may be in an enum, that can help the developper know what kind of message can be built/parsed I guess

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+              new BootNotificationResponse(LocalDateTime.now().toString(),
+                      5,
+                      RegistrationStatus.Accepted);
+      default -> throw new IllegalArgumentException("Message not recognized: "
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java:22]: if you put an enum, no more default case

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2;
+
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.OcppMessage;
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.RegistrationStatus;
+
+/**
+ * Defines a response to a BootNotification message.
+ *
+ * @param currentTime String.
+ * @param interval An int.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:10]: what is the meaning of this int ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+ * @param status RegistrationStatusEnumType.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:11]: poor documentation

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java [hunk 10]
```diff
@@ -0,0 +1,10 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2;
+
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.data.SetVariableDataType;
+import java.util.List;
+
+/**
+ * A message sent to the machine to change the configuration in OCPP 2.0.
+ */
+public record ChangeConfigurationRequest(List<SetVariableDataType> list) {
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java:9]: missing doc for List<SetVariableDataType> list
where is checkstyle when we need it ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Builds an OCPP 2.0.1 message.
+ */
+public class OcppMessageBuilder2 implements OcppMessageBuilder {
+
+  @Override
+  public OcppMessage buildMessage(WebSocketRequestMessage webSocketRequestMessage) {
+    return switch (webSocketRequestMessage.messageName()) {
+      case "BootNotification" ->
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java:19]: enum here also

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java [hunk 12]
```diff
@@ -0,0 +1,9 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.data;
+
+/**
+ * Defines a component of the machine.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java:4]: what is a component of the machine ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/RegistrationStatusEnumType.java [hunk 13]
```diff
@@ -0,0 +1,12 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.data;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import fr.uge.chargepointconfiguration.chargepoint.ocpp.RegistrationStatus;
+
+/**
+ * For the OCPP 2.0 BootNotificationResponse.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/RegistrationStatusEnumType.java:7]: ok but what does this class represent ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java [hunk 14]
```diff
@@ -0,0 +1,13 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.data;
+
+/**
+ * Defines the type used in the ChangeConfigurationSRequest.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java:4]: "ChangeConfiguration**S**Request" spelling mistake ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+ *
+ * @param attributeValue The given value to the variable.
+ * @param component The variable's component.
+ * @param variableType The component.
+ */
+public record SetVariableDataType(String attributeValue,
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java:10]: is it like a (type : key -> value) triplet for the configuration of the fields ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java [hunk 16]
```diff
@@ -0,0 +1,9 @@
+package fr.uge.chargepointconfiguration.chargepoint.ocpp.ocpp2.data;
+
+/**
+ * Defines the variable of the machine's component.
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java:4]: Is variable the name of the field you want to set ?

**Review by Adrien-COSTANDI — CHANGES_REQUESTED** [review]
missing explicit doc, I don't know many things

also there are no tests

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java (hunk 1 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java:41]: Yes, will do !

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java (hunk 2 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java:53]: We should ask @Mamar-dev what to do.
We could throw an exception or maybe silently disconnect the machine ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java (hunk 3 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java:62]: Yes, I was tired when writing this code, will do

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java (hunk 6 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java:18]: I like your idea, this way, we don't handle strings like dummies.
Will do

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java (hunk 8 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:10]: Defines the interval between two BootNotificationRequest sent by the machine.
Will add this for javadoc

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java (hunk 10 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java:9]: Checkstyle is dead
Will add

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java (hunk 9 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:11]: Yeah, I know :(

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java (hunk 12 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java:4]: I don't really know, the OCPP 2.0 documentation is not really clear on this type. Will search

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java (hunk 14 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java:4]: Oh yeah my bad, it should be SetVariablesRequest...

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java (hunk 15 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java:10]: According to the documentation, yes !
AttributeValue is the value
component is the key
variableType is the type.

This is more clear on this class, like you've said it is a triplet so the javadoc should tell you that component is the key

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java (hunk 16 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java:4]: Yeah, basically.
Will change

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java (hunk 1 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java]: Added a better javadoc, sees commit [3ec09c8](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/3ec09c81be31464014271181948424259069b7da)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java (hunk 2 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ChargePointManager.java]: Removed the IllegalArgumentException, I don't know why is it used !

See commit [c382bd1](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/c382bd1766b2e4fd1acf152f10bb4721ea487f13)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java (hunk 3 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java]: Done

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java (hunk 4 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java:20]: Done
[8234ec8](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/8234ec8dcd4ed5ec1a674e8e9be9cc1c47bf726b)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java (hunk 5 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/OcppMessageBuilder.java:28]: See [8234ec8](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/8234ec8dcd4ed5ec1a674e8e9be9cc1c47bf726b)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java (hunk 8 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:10]: Added [418f5d6](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/418f5d60aaa3566fe2f766cb1ab37295e731dadc)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java (hunk 9 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java]: Done [a5356b6](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/a5356b619c9c3cef0fe02b344ccaa9fdf89ba0bc)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java (hunk 10 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/ChangeConfigurationRequest.java:9]: This class is totally incorrect, I deleted it
This should be in another task !

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java (hunk 12 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/Component.java]: I deleted it

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/RegistrationStatusEnumType.java (hunk 13 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/RegistrationStatusEnumType.java]: Deleted

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java (hunk 14 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java]: Deleted

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java (hunk 15 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/SetVariableDataType.java]: Deleted

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java (hunk 16 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/data/VariableType.java]: Deleted

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java (hunk 6 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java]: Done

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java (hunk 7 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java]: Done

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java (hunk 11 above)
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java]: Done

**Review by Mamar-dev — CHANGES_REQUESTED** [review]

File: src/main/java/fr/uge/chargepointconfiguration/ChargepointconfigurationApplication.java [hunk 17]
```diff
@@ -50,7 +54,8 @@ public void run(String... args) throws Exception {
       Thread.ofPlatform().start(() -> {
         WebSocketServer server = new ConfigurationServer(
                 new InetSocketAddress(websocketUrl, websocketPort),
-                userRepository
+                userRepository,
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/ChargepointconfigurationApplication.java]: Done
[b793735](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/b793735f1d45a02c9a548045e0607162fa959fa9)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java [hunk 18]
```diff
@@ -37,7 +42,7 @@ public void onOpen(WebSocket conn, ClientHandshake handshake) {
     var ocppVersion = OcppVersion.parse(handshake.getFieldValue("Sec-Websocket-Protocol"));
     chargePoints.put(conn.getRemoteSocketAddress(),
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ConfigurationServer.java]: Done
[c1a91a3](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/c1a91a34a9824b6accb6c40c1bbeb5030c704275)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java [hunk 19]
```diff
@@ -0,0 +1,37 @@
...
+  public Optional<OcppMessage> buildMessage(WebSocketRequestMessage webSocketRequestMessage) {
+    return switch (webSocketRequestMessage.messageName()) {
+      case BOOT_NOTIFICATION_REQUEST ->
+              Optional.of(new BootNotificationResponse(LocalDateTime.now().toString(),
+                      5,
+                      RegistrationStatus.Accepted));
+      case UPDATE_FIRMWARE_RESPONSE ->
+              Optional.empty(); // TODO : Listen until we receive a status firmware request
+      case STATUS_FIRMWARE_REQUEST ->
+              Optional.empty(); // TODO : Send a status firmware response
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageBuilder16.java:26]: Added TODO
[12563eb](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/12563eba973b4e9f3961339c4594695832c6b7e0)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java [hunk 20]
```diff
@@ -0,0 +1,38 @@
...
+
+  @Override
+  public Optional<OcppMessage> buildMessage(WebSocketRequestMessage webSocketRequestMessage) {
+    return switch (webSocketRequestMessage.messageName()) {
+      case BOOT_NOTIFICATION_REQUEST ->
+              Optional.of(new BootNotificationResponse(LocalDateTime.now().toString(),
+                      5,
+                      RegistrationStatus.Accepted));
+      case UPDATE_FIRMWARE_RESPONSE ->
+              Optional.empty(); // TODO : Listen until we receive a status firmware request
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/OcppMessageBuilder2.java:25]: Added TODO
[12563eb](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/12563eba973b4e9f3961339c4594695832c6b7e0)

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/WebSocketRequestMessage.java [hunk 21]
```diff
@@ -23,7 +52,7 @@ public static WebSocketRequestMessage parse(String message) {
     var array = message.substring(1, message.length() - 1).split(",", 4);
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/main/java/fr/uge/chargepointconfiguration/chargepoint/WebSocketRequestMessage.java:53]: Todo in another branch
[da904b9](https://github.com/LastProject-ESIEE/OCPPConfigurationServer/pull/24/commits/da904b9dbb171f7afacbdeb04caea7672fae94ec)

File: src/test/java/fr/uge/chargepointconfiguration/ChargePointManagerTest.java [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+    var actualResponse = manager.processMessage(receivedMessage);
+    var resp = ocppMessageBuilder2.buildMessage(receivedMessage);
+    var expectedWebSocketResponseMessage = new WebSocketResponseMessage(3,
+            receivedMessage.messageId(),
+            JsonParser.objectToJsonString(resp));
+    assertEquals(expectedWebSocketResponseMessage.messageId(), actualResponse.orElseThrow().messageId());
+    assertEquals(expectedWebSocketResponseMessage.callType(), actualResponse.orElseThrow().callType());
+    // WE DON'T COMPARE THE DATE BECAUSE WE CANNOT HAVE THE SAME DATE, MS DIFFERENCE.
+  }
+
```
[the remark this answers is not in the document]
Comment (tvelu77) [reply src/test/java/fr/uge/chargepointconfiguration/ChargePointManagerTest.java:94]: For now, the test are not important, I will create an another branch with better tests

**Review by Mamar-dev — APPROVED** [review]
Good

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageParser16.java [hunk 23]
```diff
@@ -4,20 +4,28 @@
...
 
 /**
  * Parses an OCPP 1.6 message.
  */
 public class OcppMessageParser16 implements OcppMessageParser {
 
   @Override
-  public OcppMessage parseMessage(WebSocketRequestMessage webSocketRequestMessage) {
+  public Optional<OcppMessage> parseMessage(WebSocketRequestMessage webSocketRequestMessage) {
     return switch (webSocketRequestMessage.messageName()) {
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp16/OcppMessageParser16.java:16]: messageType() ?

File: src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java [hunk 24]
```diff
@@ -0,0 +1,21 @@
...
+ * to confirm that we had read the message.<br>
+ * See this as an ACK packet.
+ *
+ * @param currentTime The current server time of the server in String.
+ * @param interval This value defines the interval in second that the remote should respect
+ *                 before sending a HeartBeat packet (which is like a PING).
+ * @param status This enum serves as a response for the message. If it is accepted,
+ *               rejected or something else.
+ */
+public record BootNotificationResponse(String currentTime,
```
Comment (Adrien-COSTANDI) [inline src/main/java/fr/uge/chargepointconfiguration/chargepoint/ocpp/ocpp2/BootNotificationResponse.java:18]: could be a good idea to know which version of OCPP in the name

File: src/test/java/fr/uge/chargepointconfiguration/ChargePointManagerTest.java [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+  private final OcppMessageBuilder ocppMessageBuilder16 = OcppMessageBuilder.instantiateFromVersion(ocppVersion16);
+  private final OcppMessageBuilder ocppMessageBuilder2 = OcppMessageBuilder.instantiateFromVersion(ocppVersion2);
+  @Mock
+  private ChargepointRepository chargepointRepository;
+  @Mock
+  private WebSocket webSocket;
+
+  @Test
+  public void chargePointManager16CorrectConstructor() {
+    new ChargePointManager(ocppVersion16,
```
Comment (Adrien-COSTANDI) [inline src/test/java/fr/uge/chargepointconfiguration/ChargePointManagerTest.java:36]: assertDoesNotThrow

**Review by Adrien-COSTANDI — APPROVED** [review]
what are the mocks for ? To be continued...
GG

**Pull request merged**